Add instructions for running e2e tests (#25577)
gatsbyjs/gatsby: docs/contributing/setting-up-your-local-dev-environment.md

  - To run tests for a single package you can run: `yarn jest <package-name>`.
  - To run a single test file you can run: `yarn jest <file-path>`.

If you're adding e2e tests and want to run them against local changes:

> - In the root of the mono repo run `yarn lerna run build --scope=<package-name>` where `package-name` is the changes you're testing

Reviewer: ```suggestion
- In the root of the monorepo, run `yarn lerna run build --scope=<package-name>` where `package-name` is the directory containing the changes you're testing.
```

I found "is the changes you're testing" a little bit awkward. Feel free to fiddle further! 💖

> - Run `gatsby-dev` inside your specific e2e test directory, for example `e2e-tests/themes/development-runtime`

Reviewer: ```suggestion
- Run `gatsby-dev` inside your specific e2e test directory, for example `e2e-tests/themes/development-runtime`.
```

> - While the previous step is running open a new terminal window and run `yarn test` in that same e2e test directory

Reviewer: ```suggestion
- While the previous step is running, open a new terminal window and run `yarn test` in that same e2e test directory.
```

You and these commas! 😅

Author: I hate commas, ok!
